cli/src/main.rs

```diff
@@ -0,0 +1,188 @@
+
+use error::CliError;
+
+const APP_NAME: &str = env!("CARGO_PKG_NAME");
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+fn main() {
+    let matches = clap_app!(myapp =>
+        (name: APP_NAME)
+        (version: VERSION)
+        (author: "Cargill")
+        (about: "Cylinder CLI")
+        (@arg verbose: -v +multiple +global "Log verbosely")
+        (@arg quiet: -q --quiet +global "Do not display output")
+        (@setting SubcommandRequiredElseHelp)
+
+        (@subcommand jwt =>
+            (about: "Subcommands for generating and examining Cylinder JSON web tokens")
+            (@setting SubcommandRequiredElseHelp)
+            (@subcommand generate =>
+            (about: "Generates a JWT for a given private key")
+             (@arg private_key_file: -k --key +takes_value
+                        "Name or path of private key")
+            )
+        )
+
+    )
+    .get_matches();
+
+    // set default to info
+    let log_level = if matches.is_present("quiet") {
+        log::LevelFilter::Error
+    } else {
+        match matches.occurrences_of("verbose") {
+            0 => log::LevelFilter::Info,
+            1 => log::LevelFilter::Debug,
+            _ => log::LevelFilter::Trace,
+        }
+    };
+
+    let log_spec = LogSpecBuilder::new().default(log_level).build();
+
+    match Logger::with(log_spec)
+        .format(log_format)
+        .log_to_stdout()
+        .start()
+    {
+        Ok(_) => {}
+        #[cfg(test)]
+        // `FlexiLoggerError::Log` means the logger has already been initialized; this will happen
+        // when `run` is called more than once in the tests.
+        Err(flexi_logger::FlexiLoggerError::Log(_)) => {}
+        Err(err) => panic!("Failed to start logger: {}", err),
+    }
+
+    let res = match matches.subcommand() {
+        ("jwt", Some(matches)) => handle_jwt_subcommands(matches),
+        // Clap will have caught any unrecognized subcommands
+        (_, _) => unreachable!(),
+    };
+
+    if let Err(err) = res {
+        error!("{}", err);
+    }
+}
+
+fn handle_jwt_subcommands(matches: &clap::ArgMatches) -> Result<(), CliError> {
+    match matches.subcommand() {
+        ("generate", Some(matches)) => handle_jwt_generate(matches),
+        // Clap will have caught any unrecognized subcommands
+        (_, _) => unreachable!(),
+    }
+}
+
+fn handle_jwt_generate(matches: &clap::ArgMatches) -> Result<(), CliError> {
+    let key_name = matches.value_of("private_key_file");
+
+    let private_key = load_private_key(key_name)?;
+
+    let context = Secp256k1Context::new();
+    let signer = context.new_signer(private_key);
+
+    let encoded_token = JsonWebTokenBuilder::new().build(&*signer).map_err(|err| {
+        CliError::from_source_with_message(Box::new(err), "failed to build json web token".into())
+    })?;
+
+    info!("{}", encoded_token);
+
+    Ok(())
+}
+
+fn load_private_key(key_name: Option<&str>) -> Result<PrivateKey, CliError> {
```

Comment:
We shouldn't need this wrapper, we can just use the cylinder functions directly since we don't need to worry about any non-cylinder directories.